packages/typespec-ts/test/unit/modelsGenerator.spec.ts

```diff
@@ -712,14 +712,110 @@ describe("Input/output model type", () => {
+        describe("is string", () => {
+          const typespec = (hasDiscriminator: boolean) => `
+          enum A {
+            AA,
+            BB,
+          }
+          ${hasDiscriminator ? '@discriminator("a")' : ""}
+          model B {
+            a: A,
+          } 
+          model C extends B {
+            a: A.AA,
+          }
+          op read(): { @body body: C };
+          `;
+          it("with @discriminator", async () => {
+            const schemaOutput = await emitModelsFromCadl(typespec(true));
+            assert.ok(schemaOutput);
+            const { inputModelFile, outputModelFile } = schemaOutput!;
+            assert.ok(!inputModelFile?.content);
+            assert.strictEqual(outputModelFile?.path, "outputModels.ts");
+            assertEqualContent(
+              outputModelFile?.content!,
+              `
+            export interface COutput extends BOutputParent {
+              a: "AA";
+            }
+
+            export interface BOutputParent {
+              a: string;
+            }
+
+            export type BOutput = COutput;`
+            );
+          });
+
+          it("without @discriminator", async () => {
+            const schemaOutput = await emitModelsFromCadl(typespec(false));
+            assert.ok(schemaOutput);
+            const { inputModelFile, outputModelFile } = schemaOutput!;
+            assert.ok(!inputModelFile?.content);
+            assert.strictEqual(outputModelFile?.path, "outputModels.ts");
+            assertEqualContent(
+              outputModelFile?.content!,
+              `
+            export interface COutput extends BOutput {
+              a: "AA";
+            }
+    
+            export interface BOutput {
+              /** Possible values: AA, BB */
+              a: string;
+            }`
+            );
+          });
+        });
+
+        describe("is number", () => {
+          const typespec = (hasDiscriminator: boolean) => `
+          enum A {
+            AA: 1.1,
+            BB: 2.2,
+          }
+          ${hasDiscriminator ? '@discriminator("a")' : ""}
+          model B {
+            a: A,
+          }
+          model C extends B {
+            a: A.AA,
+          }
+          op read(): { @body body: C };
+          `;
+
+          it("without @discriminator", async () => {
+            const schemaOutput = await emitModelsFromCadl(typespec(false));
+            assert.ok(schemaOutput);
+            const { inputModelFile, outputModelFile } = schemaOutput!;
+            assert.ok(!inputModelFile?.content);
+            assert.strictEqual(outputModelFile?.path, "outputModels.ts");
+            assertEqualContent(
+              outputModelFile?.content!,
+              `
+            export interface COutput extends BOutput {
+              a: 1.1;
+            }
+    
+            export interface BOutput {
+              /** Possible values: 1.1, 2.2 */
+              a: string;
```

Comment:
Should this be number in this case ?

Reply:
number type is not allowed for tsp compiler so we would always fail in this case and in UT we have no way to catch this error.